### ISSUE TITLE
MongoDB $unwind is not supported

### ISSUE THREAD

**Issue by dreampper** [Azure/azure-documentdb-dotnet#234]
Im trying to use $unwind in my command and I'm getting the error "$unwind is not supported".

It was introduced in Mongo 3.2. Any idea of when will it work with DocumentDB?

**Comment by sivethe** [comment]
@dreampper:  As the message says, we currently don't support $unwind and $group stages.  We are working on adding support for these stages and it should be available by Q2 2017.

**Comment by nmarjanovic** [comment]
Is this done, we have issue with this as well?

**Comment by rishikeshjadhav** [comment]
Any update on this?

**Comment by nmarjanovic** [comment]
@rishikeshjadhav 

For people looking for solution, just contact Azure support and they can enable $group support, we are testing right now, but it's too soon to say if this beta solution going to work as native Mongodb solution.

**Comment by rcarmo** [comment]
Well, this would have come in very handy right now, for doing aggregation on sub-documents. Time to re-think my entire schema...

**Comment by AndrewHoh** [comment]
@rcarmo Can you reach out to me at anhoh{at}microsoft{dot}com? We might be able to help out.

**Comment by nmarjanovic** [comment]
We are moving from CosmosDB to managed native mongodb, new errors pop up with other parts of our application, don't have time to spend on sending ticket after ticket to Microsoft and get answer after 24h or more. Maybe in 1 year or 2 from now, wen they will run pure mongodb service and stop using clients as their quality test service. I'm not rude, but they need to work to many things and that is not what we are going to use for our product in production environment. Azure ok, their base, no thanks.

**Comment by dpfavand** [comment]
@sivethe any update on $unwind and $group now in Q4 2017? I've been very happy with CosmosDB but the lack of aggregate functions is an unexpected wall, particularly given that the docs indicate complete compatibility.

**Comment by odinnou** [comment]
yep... "$group is not supported" is the reason why I'll migrate to another mongoDB Azure solution

**Comment by sivethe** [comment]
@dpfavand @odinnou :  We now support $group/$unwind and most of the other aggregation pipeline operators in private preview (soon to be public preview).  Please follow the steps at https://feedback.azure.com/forums/599059-azure-cosmos-db-mongodb-api/suggestions/19449010-aggregation-pipeline to enable this feature on your account
